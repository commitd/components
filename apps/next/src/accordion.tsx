import {
  Accordion,
  AccordionContent,
  AccordionHeader,
  AccordionItem,
} from '@committed/ds'

export const AccordionExample = () => (
  <Accordion type="multiple">
    <AccordionItem value="item-1">
      <AccordionHeader>Item 1</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 1.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2">
      <AccordionHeader>Item 2</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 2.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-3">
      <AccordionHeader>Item 3</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 3.
      </AccordionContent>
    </AccordionItem>
  </Accordion>
)
