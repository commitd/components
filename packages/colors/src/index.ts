export { presetColors } from './preset'
export type { Color } from './preset'
