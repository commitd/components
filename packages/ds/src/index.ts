// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
