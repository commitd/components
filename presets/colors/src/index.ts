export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
