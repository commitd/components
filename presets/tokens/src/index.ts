export { preset as presetTokens } from './preset'
