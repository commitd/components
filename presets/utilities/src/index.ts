export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

