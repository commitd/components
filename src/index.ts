export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
