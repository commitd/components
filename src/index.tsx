export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
