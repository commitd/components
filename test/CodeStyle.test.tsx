import React from 'react'
import { renderLight, renderDark } from './setupTests'
import { CodeStyle } from '../src'

const Example = () => (
  <CodeStyle>
    <div
      dangerouslySetInnerHTML={{
        __html:
          '<pre class="language-json"><code class="language-json"><span class="token punctuation">{</span>\n' +
          '  <span class="token property">"scripts"</span><span class="token operator">:</span> <span class="token punctuation">{</span>\n' +
          '    <span class="token property">"format"</span><span class="token operator">:</span> <span class="token string">"prettier --write \'{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}\'"</span><span class="token punctuation">,</span>\n' +
          '    <span class="token property">"format-check"</span><span class="token operator">:</span> <span class="token string">"prettier --check \'{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}\'"</span>\n' +
          '  <span class="token punctuation">}</span><span class="token punctuation">,</span>\n' +
          '  <span class="token property">"prettier"</span><span class="token operator">:</span> <span class="token punctuation">{</span>\n' +
          '    <span class="token property">"semi"</span><span class="token operator">:</span> <span class="token boolean">false</span><span class="token punctuation">,</span>\n' +
          '    <span class="token property">"singleQuote"</span><span class="token operator">:</span> <span class="token boolean">true</span>\n' +
          '  <span class="token punctuation">}</span><span class="token punctuation">,</span>\n' +
          '  <span class="token property">"husky"</span><span class="token operator">:</span> <span class="token punctuation">{</span>\n' +
          '    <span class="token property">"hooks"</span><span class="token operator">:</span> <span class="token punctuation">{</span>\n' +
          '      <span class="token property">"pre-commit"</span><span class="token operator">:</span> <span class="token string">"pretty-quick --staged"</span>\n' +
          '    <span class="token punctuation">}</span>\n' +
          '  <span class="token punctuation">}</span>\n' +
          '<span class="token punctuation">}</span>\n' +
          '</code></pre>',
      }}
    />
  </CodeStyle>
)

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<Example />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<Example />)
  expect(asFragment()).toMatchSnapshot()
})
