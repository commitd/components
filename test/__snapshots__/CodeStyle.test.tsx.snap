// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <div
    class="div-root div-root"
  >
    <div>
      <pre
        class="language-json"
      >
        <code
          class="language-json"
        >
          <span
            class="token punctuation"
          >
            {
          </span>
          
  
          <span
            class="token property"
          >
            "scripts"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "format"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "prettier --write '{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}'"
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
    
          <span
            class="token property"
          >
            "format-check"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "prettier --check '{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}'"
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
  
          <span
            class="token property"
          >
            "prettier"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "semi"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token boolean"
          >
            false
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
    
          <span
            class="token property"
          >
            "singleQuote"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token boolean"
          >
            true
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
  
          <span
            class="token property"
          >
            "husky"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "hooks"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
      
          <span
            class="token property"
          >
            "pre-commit"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "pretty-quick --staged"
          </span>
          
    
          <span
            class="token punctuation"
          >
            }
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          

          <span
            class="token punctuation"
          >
            }
          </span>
          

        </code>
      </pre>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <div
    class="div-root div-root"
  >
    <div>
      <pre
        class="language-json"
      >
        <code
          class="language-json"
        >
          <span
            class="token punctuation"
          >
            {
          </span>
          
  
          <span
            class="token property"
          >
            "scripts"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "format"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "prettier --write '{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}'"
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
    
          <span
            class="token property"
          >
            "format-check"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "prettier --check '{**/*.js,*.ts,*.tsx,*.json,*.md,*.mdx}'"
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
  
          <span
            class="token property"
          >
            "prettier"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "semi"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token boolean"
          >
            false
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
    
          <span
            class="token property"
          >
            "singleQuote"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token boolean"
          >
            true
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          <span
            class="token punctuation"
          >
            ,
          </span>
          
  
          <span
            class="token property"
          >
            "husky"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
    
          <span
            class="token property"
          >
            "hooks"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token punctuation"
          >
            {
          </span>
          
      
          <span
            class="token property"
          >
            "pre-commit"
          </span>
          <span
            class="token operator"
          >
            :
          </span>
           
          <span
            class="token string"
          >
            "pretty-quick --staged"
          </span>
          
    
          <span
            class="token punctuation"
          >
            }
          </span>
          
  
          <span
            class="token punctuation"
          >
            }
          </span>
          

          <span
            class="token punctuation"
          >
            }
          </span>
          

        </code>
      </pre>
    </div>
  </div>
</DocumentFragment>
`;
